crates/matrix-sdk/src/send_queue/mod.rs

```diff
@@ -236,7 +236,20 @@ impl SendQueue {
 
     /// Get or create a new send queue for a given room, and insert it into our
     /// memoized rooms mapping.
+    #[cfg(not(test))]
     fn for_room(&self, room: Room) -> RoomSendQueue {
+        self.for_room_impl(room)
+    }
+
+    /// Get or create a new send queue for a given room, and insert it into our
+    /// memoized rooms mapping.
+    #[cfg(test)]
+    pub(crate) fn for_room(&self, room: Room) -> RoomSendQueue {
+        self.for_room_impl(room)
+    }
+
+    #[inline(always)]
+    fn for_room_impl(&self, room: Room) -> RoomSendQueue {
```

Comment:
Oof, it's fine to mark the initial `for_room` as `pub(crate)` IMO, especially if it's only for use in testing. It confused me and took me 15 seconds to find the difference between the two impls here…

Reply:
👍